### ISSUE TITLE
How to integrate the modbus-go for Kubeedge 1.4/ Need some documents  

### ISSUE THREAD

**Issue by mxtsuper** [kubeedge/kubeedge#2384]
**What happened**:
I am trying to use the modbus-go for Kubeedge 1.4.   I appreciate some documents or guidings on this process. 
I found some doc. on modbus_mapper, which is only for Kubeedge 1.3 and before.

I have some issues on integrating the modbus-go. Please give me some hints.

I also need some guides how to test the modbus go ? 
**What you expected to happen**:

I was building the docker image with the Dockerfile in the modbus-go folder.  
Then it showed some errors in step 3/6. 
"COPY failed: stat /var/lib/docker/tmp/docker-builder424651524/modbus: no such file or directory"

It can not find the modbus file in the folder. 
How can I find or build the modbus file?

Without the docker images, I can not create the deployment. The modbus pod was always in pending status.
 
**How to reproduce it (as minimally and precisely as possible)**:
Enter into modbus-go folder.
docker build -t modbusmapper:v1.0 .

**Anything else we need to know?**:

**Environment**:
<details><summary>Kubernets Version</summary>

```console
$ kubelet --version
# paste output here
$ kubectl version
# 1.17
```

</details>

<details><summary>CloudCore Version</summary>

```console
$ cloudcore --version
# 1.14
```

</details>

<details><summary>EdgeCore Version</summary>

```console
$ edgecore --version
# paste output here
``` 1.14

</details>

**CloudSide Environment**:
<details><summary>Hardware configuration</summary>

```console
$ lscpu
# paste output here
```

</details>

<details><summary>OS</summary>

```console
$ cat /etc/os-release
# paste output here
```

</details>

<details><summary>Kernel</summary>

```console
$ uname -a
# paste output here
```

</details>

<details><summary>Go version</summary>

```console
$ go version
# paste output here
```

</details>

<details><summary>Others</summary>
</details>

**EdgeSide Environment**:
<details><summary>Hardware configuration</summary>

```console
$ lscpu
# paste output here
```

</details>

<details><summary>OS</summary>

```console
$ cat /etc/os-release
# paste output here
```

</details>

<details><summary>Kernel</summary>

```console
$ uname -a
# paste output here
```

</details>

<details><summary>Go version</summary>

```console
$ go version
# paste output here
```

</details>

<details><summary>Others</summary>
</details>

**Comment by daixiang0** [comment]
Modbus-go is for KubeEdge 1.5 refer to https://github.com/kubeedge/kubeedge/blob/master/CHANGELOG/CHANGELOG-1.5.md.

**Comment by mxtsuper** [comment]
The modubs-go ReadMe tells the messages. 
Mappers
There're two folders for modbus mapper. "modbus-go" is written by go language, "modbus_mapper" is by JavaScript.

Note, the modbus-go could work for KubeEdge 1.4 version and later. modbus_mapper is only for KubeEdge 1.3 and before.

**Comment by sailorvii** [comment]
You could make the image in kubeedge/kubeedge with command:
make modbusmapper_image

or 
sudo docker build -t modbusmapper:v1.0 ./mappers/modbus-go

**Comment by mxtsuper** [comment]
I fixed the modbus issues finally. 
My solutions: 
1. Besides the modbus deployment guides, I created a configmap with the deviceprofile file.
2. I found and fixed some bugs in the codes. There was some mismatches in the device/device model and modbus.js stacks.
    The kubeedge team has fixed the bugs so far.
3. The modbus.go stack can run on the kubeedge 1.4 version. I ran it successfully.

Thanks for your feedbacks and supports.
